### PR TITLE
Feat(eos_cli_config_gen): Add knob to define port in GNMI transport grpc

### PR DESCRIPTION
**Pull request by philippebureau**
## Change Summary

Add the capacity to define a custom port in management api gnmi for transport grpc

## Related Issue(s)

N/A

## Component(s) name

`arista.avd.eos_cli_config_gen`

## How to test
see molecule tests :
    management-gnmi.yml
    management-gnmi-new-flags.yml

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/management-gnmi-new-flags.md [hunk 1]
```diff
@@ -56,6 +56,7 @@ management api gnmi
       ip access-group acl1
       notification timestamp send-time
    transport grpc mytransport
+      port 6032
       ip access-group acl1
       notification timestamp send-time
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/management-gnmi-new-flags.md:61]: please verify the order on EOS.

Comment (philippebureau) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/management-gnmi-new-flags.md:61]: Yes it is in order

**Comment by noredistribution** [comment]
should we update the documentation? changing the gRPC Port might require update to the control-plane ACL

**Comment by philippebureau** [comment]
I can add a note to the description for sure, give me 1 h

**Review by noredistribution — COMMENTED** [review]
tested and works fine for me, I'd just add that note about the CP ACL and then it's good to go

**Comment by philippebureau** [comment]
> should we update the documentation? changing the gRPC Port might require update to the control-plane ACL

updated schema fragment description with your recommendation. Thank you

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_api_gnmi.schema.yml [hunk 2]
```diff
@@ -42,6 +42,11 @@ keys:
                 ip_access_group:
                   type: str
                   description: ACL name
+                port:
+                  type: int
+                  description: GNMI port.  Make sure to update the control-plane ACL accordingly in order for the service to be reachable by external applications.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_api_gnmi.schema.yml]: Cosmetic:

```suggestion
                  description: |
                      GNMI port. 
                      Make sure to update the control-plane ACL accordingly in order for the service to be reachable by external applications.
```

**Review by gmuloc — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/management-gnmi-new-flags.md [hunk 3]
```diff
@@ -38,10 +38,10 @@ interface Management1
...
 #### Management API gNMI Summary
 
-| Transport | SSL Profile | VRF | Notification Timestamp | ACL |
-| --------- | ----------- | --- | ---------------------- | --- |
-| MGMT | gnmi | MGMT | send-time | acl1 |
-| mytransport | - | - | send-time | acl1 |
+| Transport | SSL Profile | VRF | Notification Timestamp | ACL | Port |
+| --------- | ----------- | --- | ---------------------- | --- | ---- |
+| MGMT | gnmi | MGMT | send-time | acl1 | 6030
+| mytransport | - | - | send-time | acl1 | 6032
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/management-gnmi-new-flags.md]:
```suggestion
| MGMT | gnmi | MGMT | send-time | acl1 | 6030 |
| mytransport | - | - | send-time | acl1 | 6032 |
```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/management-gnmi.md [hunk 4]
```diff
@@ -38,10 +38,10 @@ interface Management1
...
 #### Management API gNMI Summary
 
-| Transport | SSL Profile | VRF | Notification Timestamp | ACL |
-| --------- | ----------- | --- | ---------------------- | --- |
-| MGMT | - | MGMT | last-change-time | ACL-GNMI |
-| MONITORING | - | MONITORING | last-change-time | - |
+| Transport | SSL Profile | VRF | Notification Timestamp | ACL | Port |
+| --------- | ----------- | --- | ---------------------- | --- | ---- |
+| MGMT | - | MGMT | last-change-time | ACL-GNMI | 6030
+| MONITORING | - | MONITORING | last-change-time | - | 6031
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/management-gnmi.md]:
```suggestion
| MGMT | - | MGMT | last-change-time | ACL-GNMI | 6030 |
| MONITORING | - | MONITORING | last-change-time | - | 6031 |
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/management-api-gnmi.j2 [hunk 5]
```diff
@@ -26,16 +26,17 @@
...
 {%         for transport in management_api_gnmi.transport.grpc %}
 {%             if transport.name is arista.avd.defined %}
 {%                 set transport_name = transport.name %}
 {%                 set ssl_profile = transport.ssl_profile | arista.avd.default('-') %}
 {%                 set vrf = transport.vrf | arista.avd.default('-') %}
 {%                 set notif = transport.notification_timestamp | arista.avd.default('last-change-time') %}
 {%                 set acl = transport.ip_access_group | arista.avd.default('-') %}
-| {{ transport_name }} | {{ ssl_profile }} | {{ vrf }} | {{ notif }} | {{ acl }} |
+{%                 set port = transport.port | arista.avd.default('6030') %}
+| {{ transport_name }} | {{ ssl_profile }} | {{ vrf }} | {{ notif }} | {{ acl }} | {{ port }}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/management-api-gnmi.j2]:
```suggestion
| {{ transport_name }} | {{ ssl_profile }} | {{ vrf }} | {{ notif }} | {{ acl }} | {{ port }} |
```

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

**Review by ClausHolbechArista — APPROVED** [review]

**Pull request merged**